docs/pages/usage.md

```diff
@@ -0,0 +1,124 @@
+est = EuclideanKNNRegressor(n_neighbors=3).fit(X_train, y_train)
+
+print(est.score(X_test, y_test)) 
+# 0.11496218649569434
+```
+
+In addition to their core functionality of fitting, predicting, and scoring, `sknnr` estimators offer a number of other features, detailed below.
+
+### Independent Scores and Predictions
+
+When an independent test set is not available, the accuracy of a kNN regressor can be estimated by comparing each sample in the training set to its second-nearest neighbor, i.e. the closest point *excluding itself*. All `sknnr` estimators set `independent_prediction_` and `independent_score_` attributes when they are fit, which store the predictions and scores of this independent evaluation.
+
+```python
+print(est.independent_score_)
+# 0.10243925752772305
+```
+
+### Retrieving Dataframe Indexes
+
+In `sklearn`, the `KNeighborsRegressor.kneighbors` method can identify the array index of the nearest neighbor to a given sample. Estimators in `sknnr` offer an additional parameter `return_dataframe_index` that allows neighbor plots to be identified directly by their index.
+
+```python
+X, y = load_swo_ecoplot(return_X_y=True, as_frame=True)
+est = est.fit(X, y)
+
+# Find the distance and dataframe index of the nearest neighbors to the first plot
+distances, neighbor_ids = est.kneighbors(X.iloc[:1], return_dataframe_index=True)
+
+# Preview the nearest neighbors by their dataframe index
+print(y.loc[neighbor_ids[0]])
+```
+
+|       |   ABAM_COV |   ABGRC_COV |   ABPRSH_COV |   ACMA3_COV |   ALRH2_COV |
+|------:|-----------:|------------:|-------------:|------------:|------------:|
+| 52481 |          0 |           0 |      39.3469 |           0 |           0 |
+| 60089 |          0 |           0 |      22.1199 |           0 |           0 |
+| 56253 |          0 |           0 |      22.8948 |           0 |           0 |
+
+!!! warning
+    An estimator must be fit with a `DataFrame` in order to use `return_dataframe_index=True`.
+
+### Y-Fit Data
+
+The [GNNRegressor](api/estimators/gnn.md) and [MSNRegressor](api/estimators/msn.md) estimators can be fit with `X` and `y` data, but they also accept an optional `y_fit` parameter. If provided, `y_fit` is used to fit the ordination transformer while `y` is used to fit the kNN regressor.
+
+```python
+from sknnr import GNNRegressor
+
+est = GNNRegressor().fit(X, y, y_fit=y_fit)
+```
+
+### Dimensionality Reduction
+
+The ordination transformers used by the [GNNRegressor](api/estimators/gnn.md) and [MSNRegressor](api/estimators/msn.md) estimators apply dimensionality reduction to label data. Dimensionality can be further reduced by specifying `n_components` when instantiating the estimator.
+
+```python
+est = GNNRegressor(n_components=3).fit(X, y)
+```
+
+!!! warning
+    The maximum number of components depends on the input data and the estimator. Specifying `n_components` greater than the maximum number of components will raise an error.
```

Comment:
We could include more detail here on why the max number of components would vary.

Reply:
This is a good point.  I'm a bit leery of this for a couple of different reasons:

1. In the case of `CCA`, I personally don't know the math well enough to know when the rank of a matrix would be less than the number of covariates going in.  In the case of `CCorA`, I would need to understand what `ftest_cor` is doing to be confident in explaining it.  So a bit of imposter syndrome going on here.
2. This might get into a bit too much detail for the usage page.  I think it might be appropriate to detail what the maximum components would be on the specific transformer pages (if we can sort out the math) and then possibly link to those sections from the usage page?

For now, I'm comfortable with what you have here, but let me know if you feel differently.

Any